src/rgw/driver/rados/rgw_cr_rados.cc

```diff
@@ -938,11 +945,22 @@ int RGWContinuousLeaseCR::operate(const DoutPrefixProvider *dpp)
     last_renew_try_time = ceph::coarse_mono_clock::now();
     while (!going_down) {
       current_time = ceph::coarse_mono_clock::now();
-      yield call(new RGWSimpleRadosLockCR(async_rados, store, obj, lock_name, cookie, interval));
+      if (bid_mgr) {
+        if (shard_id == 0) {
+          yield call(bid_mgr->notify_cr());
+        }
+        if(renew || bid_mgr->is_highest_bidder(shard_id, current_time)) {
+          yield call(new RGWSimpleRadosLockCR(async_rados, store, obj, lock_name, cookie, lock_duration_secs));
+        }
+        else {
+          retcode = -EBUSY;
+        } 
+      } else {
+        yield call(new RGWSimpleRadosLockCR(async_rados, store, obj, lock_name, cookie, lock_duration_secs));
+      }
```

Comment:
i don't think that `RGWContinuousLeaseCR` should depend on the `BidManager`

these `notify_cr()` calls should happen in the background on a separate timer (i.e. a coroutine spawned by `RGWMetaSyncCR`). calling `notify_cr()` here in the renewal loop could delay a lock renewal and cause a timeout

`RGWMetaSyncShardCR` (or maybe `RGWMetaSyncShardControlCR`?) should be the one consulting the `BidManager` to decide whether or not to call `RGWContinuousLeaseCR` in the first place. once locked, it should also periodically check `is_highest_bidder()` so it can exit gracefully (after updating sync status) if we're outbid